### PR TITLE
Rework remote server

### PR DESCRIPTION
**Pull request by OznOg**
make the server foreground by default and remove the dynamic lib remote processor
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38326360%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38409283%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38410387%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38410563%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38413902%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38768104%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38842441%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%200b25f0cdf1423aefafc3f6f30301ecb42a1c6be4%20parameter/ParameterMgrPlatformConnector.cpp%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38326360%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22API%20breakage%20%28not%20intended%2C%20as%20discussed%20with%20you%29%22%2C%20%22created_at%22%3A%20%222015-08-31T15%3A53%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgrPlatformConnector.cpp%3AL179-185%22%7D%2C%20%22Pull%20ef950540c5a9471f3751c8899610b108c23e15c3%20remote-processor/BackgroundRemoteProcessorServerBuilder.h%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38410563%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22you%20aren%27t%20using%20any%20assert.%22%2C%20%22created_at%22%3A%20%222015-09-01T11%3A57%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222015-09-07T08%3A30%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/BackgroundRemoteProcessorServerBuilder.h%3AL28-72%22%7D%2C%20%22Pull%20ef950540c5a9471f3751c8899610b108c23e15c3%20parameter/CMakeLists.txt%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38409283%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22this%20comment%20is%20now%20wrong.%22%2C%20%22created_at%22%3A%20%222015-09-01T11%3A36%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222015-09-04T16%3A18%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9692938%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cc6565%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/CMakeLists.txt%3AL110-122%22%7D%2C%20%22Pull%20ef950540c5a9471f3751c8899610b108c23e15c3%20test/functional-tests/CMakeLists.txt%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38413902%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20%3F%22%2C%20%22created_at%22%3A%20%222015-09-01T12%3A44%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests/CMakeLists.txt%3AL47-54%22%7D%2C%20%22Pull%20ef950540c5a9471f3751c8899610b108c23e15c3%20remote-processor/CMakeLists.txt%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/174%23discussion_r38410387%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22If%20I%27m%20not%20mistaken%2C%20it%27s%20the%20other%20way%20around%3A%20it%20is%20needed%20for%20shared%20libs%20to%20link%20against%20this%20lib.%22%2C%20%22created_at%22%3A%20%222015-09-01T11%3A54%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/CMakeLists.txt%3AL26-46%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 0b25f0cdf1423aefafc3f6f30301ecb42a1c6be4 parameter/ParameterMgrPlatformConnector.cpp 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/174#discussion_r38326360'>File: parameter/ParameterMgrPlatformConnector.cpp:L179-185</a></b>
- <a href='https://github.com/dawagner'></a> API breakage (not intended, as discussed with you)
- [ ] <a href='#crh-comment-Pull ef950540c5a9471f3751c8899610b108c23e15c3 parameter/CMakeLists.txt 7'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/174#discussion_r38409283'>File: parameter/CMakeLists.txt:L110-122</a></b>
- <a href='https://github.com/dawagner'></a> this comment is now wrong.
- <a href='https://github.com/cc6565'></a> done
- [ ] <a href='#crh-comment-Pull ef950540c5a9471f3751c8899610b108c23e15c3 remote-processor/CMakeLists.txt 19'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/174#discussion_r38410387'>File: remote-processor/CMakeLists.txt:L26-46</a></b>
- <a href='https://github.com/dawagner'></a> If I'm not mistaken, it's the other way around: it is needed for shared libs to link against this lib.
- [ ] <a href='#crh-comment-Pull ef950540c5a9471f3751c8899610b108c23e15c3 remote-processor/BackgroundRemoteProcessorServerBuilder.h 6'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/174#discussion_r38410563'>File: remote-processor/BackgroundRemoteProcessorServerBuilder.h:L28-72</a></b>
- <a href='https://github.com/dawagner'></a> you aren't using any assert.
- <a href='https://github.com/cc6565'></a> done
- [ ] <a href='#crh-comment-Pull ef950540c5a9471f3751c8899610b108c23e15c3 test/functional-tests/CMakeLists.txt 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/174#discussion_r38413902'>File: test/functional-tests/CMakeLists.txt:L47-54</a></b>
- <a href='https://github.com/dawagner'></a> why ?

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/174?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/174?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/174'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/174'>Refresh</a>.<a href='#crh-update-none'></a>

File: parameter/ParameterMgrPlatformConnector.cpp [hunk 1]
```diff
@@ -179,12 +179,6 @@ bool CParameterMgrPlatformConnector::start(string& strError)
     return true;
 }
 
-// Started state
-bool CParameterMgrPlatformConnector::isStarted() const
```
Comment (dawagner) [inline parameter/ParameterMgrPlatformConnector.cpp]: API breakage (not intended, as discussed with you)

File: parameter/CMakeLists.txt [hunk 2]
```diff
@@ -110,9 +110,12 @@ include_directories(
     "${PROJECT_SOURCE_DIR}/remote-processor"
     "${PROJECT_SOURCE_DIR}/parameter/log/include")
 
+set(CMAKE_THREAD_PREFER_PTHREAD 1)
+find_package(Threads REQUIRED)
+
 # No need to link with libremote-processor: it is accessed via dlopen()
```
Comment (dawagner) [inline parameter/CMakeLists.txt]: this comment is now wrong.

File: remote-processor/CMakeLists.txt [hunk 3]
```diff
@@ -26,21 +26,20 @@
...
         RequestMessage.cpp
         AnswerMessage.cpp
-        RemoteProcessorServer.cpp
-        RemoteProcessorServerBuilder.cpp)
+        RemoteProcessorServer.cpp)
 
-set(CMAKE_THREAD_PREFER_PTHREAD 1)
-find_package(Threads REQUIRED)
+if (NOT WIN32)
+    # '-fPIC' needed for linking against shared libraries
```
Comment (dawagner) [inline remote-processor/CMakeLists.txt]: If I'm not mistaken, it's the other way around: it is needed for shared libs to link against this lib.

File: remote-processor/BackgroundRemoteProcessorServerBuilder.h [hunk 4]
```diff
@@ -28,13 +28,44 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 #include "RemoteProcessorServer.h"
+#include <future>
+#include <iostream>
+#include <assert.h>
```
Comment (dawagner) [inline remote-processor/BackgroundRemoteProcessorServerBuilder.h]: you aren't using any assert.

File: test/functional-tests/CMakeLists.txt [hunk 5]
```diff
@@ -47,7 +47,7 @@ if(BUILD_TESTING)
                    catch.cpp)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 
-    target_link_libraries(parameterFunctionalTest parameter)
+    target_link_libraries(parameterFunctionalTest parameter remote-processor)
```
Comment (dawagner) [inline test/functional-tests/CMakeLists.txt]: why ?

File: parameter/CMakeLists.txt (hunk 2 above)
Comment (cc6565) [inline parameter/CMakeLists.txt]: done

File: remote-processor/BackgroundRemoteProcessorServerBuilder.h (hunk 4 above)
Comment (cc6565) [inline remote-processor/BackgroundRemoteProcessorServerBuilder.h]: done

**Comment by dawagner** [comment]
Taken over by @cc6565 in #185.

**Pull request closed without merging**